src/modules/tpm/tests/TpmTests.cpp

```diff
@@ -33,36 +36,90 @@ std::string TestTpm::RunCommand(const char* command)
 
 namespace OSConfig::Platform::Tests
 {
-    const int maxPayloadSizeBytes = 4000;
+    const unsigned int maxPayloadSizeBytes = 4000;
 
     const char* tpmVersionNumber = "\"1.2\"";
-    const char* microsoftVirtualTpmVersionNumber = "\"2.0\"";
     const char* tpmManufacturerNameSTMicroelectronics = "\"STMicroelectronics\"";
-    const char* tpmManufacturerNameMicrosoft = "\"Microsoft\"";
     const char* tpmManufacturerNameLong = "\"Manufacturer name is long and contains numb3rs and $pec!@l characters\"";
     const char* empty = "";
-    const char* quotes = "\"\"";
     const char* clientName = "ClientName";
 
     const std::string tpmDeviceDirectory = "/dev/tpm0";
     const std::string tpmDetails = "Manufacturer: 0x53544d6963726f656c656374726f6e696373\n"
                                    "TCG version: 1.2\n";
     const std::string tpmDetailsLeadingAndTrailingWhitespace = "Manufacturer: 0x202053544d6963726f656c656374726f6e6963732020\n"
                                    "TCG version:   1.2  \n";
-    const std::string tpmDetailsVersionNotFound = "Manufacturer: 0x53544d6963726f656c656374726f6e696373\n";
-    const std::string tpmDetailsManufacturerNotFound = "TCG version: 1.2\n";
     const std::string tpmDetailsManufacturerNameLong =
     "Manufacturer: 0x4d616e756661637475726572206e61"
     "6d65206973206c6f6e6720616e642063"
     "6f6e7461696e73206e756d6233727320"
     "616e64202470656321406c2063686172"
     "616374657273\n"
     "TCG version: 1.2\n";
-    const std::string microsoftVirtualTpmDetails = "Microsoft Virtual TPM 2.0";
 
     const std::string tpmDetected = std::to_string(TestTpm::Status::TpmDetected);
     const std::string tpmNotDetected = std::to_string(TestTpm::Status::TpmNotDetected);
 
+    TEST(TpmTests, UnsignedInt8ToUnsignedInt64)
+    {
+        uint8_t test[8] = {0x74, 0x65, 0x73, 0x74, 0x64, 0x61, 0x74, 0x61}; // Hexadecimal representation of {'t','e','s','t','d','a','t','a'}
+        uint8_t* inputBuf = &test[0];
+        uint32_t dataOffset = 0;
+        uint32_t dataLength = 4;
+        uint64_t data = 0;
+
+        int status = MMI_OK;
+        status = Tpm2Utils::UnsignedInt8ToUnsignedInt64(nullptr, TPM_RESPONSE_MAX_SIZE, dataOffset, dataLength, &data);
+
+        EXPECT_EQ(status, EINVAL);
+
+        status = Tpm2Utils::UnsignedInt8ToUnsignedInt64(inputBuf, TPM_RESPONSE_MAX_SIZE, dataOffset, dataLength, nullptr);
+
+        EXPECT_EQ(status, EINVAL);
```

Comment:
These can be combined in one single line. Same for the other instances. The test function cab be made more compact without separator lines.